src/main/java/org/opensrp/web/rest/EventResource.java

```diff
@@ -537,6 +564,137 @@ public Event update(Event entity) {
 			return new ResponseEntity<>(INTERNAL_SERVER_ERROR);
 		}
 	}
+
+	@GetMapping(value = "/export-data", produces = "application/zip")
+	public ResponseEntity<ByteArrayResource> exportEventData(@RequestParam List<String> eventTypes,
+			@RequestParam String planIdentifier,
+			@RequestParam(value = "fromDate", required = false) String fromDate,
+			@RequestParam(value = "toDate", required = false) String toDate) throws IOException {
+
+		FileOutputStream fos = null;
+		ZipOutputStream zipOS = null;
+		String exportDataFileName;
+		String missionName = "";
+		String eventTypeName;
+		SimpleDateFormat df = new SimpleDateFormat("yyyy-MM-dd");
+		String formatted = "";
+		String zipFileName = "";
+		boolean firstTime = true;
+
+		try {
+			for (String eventType : eventTypes) {
+				ExportEventDataSummary exportEventDataSummary = eventService
+						.exportEventData(planIdentifier, eventType, Utils.getDateTimeFromString(fromDate),
+								Utils.getDateTimeFromString(toDate));
+
+				missionName = exportEventDataSummary != null &&
+						exportEventDataSummary.getMissionName() != null ?
+						exportEventDataSummary.getMissionName().replaceAll("\\s+", "_").toLowerCase() :
+						"";
+				eventTypeName = eventType.replaceAll("\\s+", "_");
+
+				formatted = df.format(new Date());
+				File csvFile = null;
+
+				zipFileName = SAMPLE_CSV_FILE + missionName + "_" + formatted + ".zip";

```

Comment:
shouldn't this be created in a temp directory

Reply:
Done